collect_app/src/main/java/org/odk/collect/android/audio/AudioPlayerViewModel.java

```diff
@@ -0,0 +1,190 @@
+            if (isCurrentPlayingClip(clipID, value)) {
+                return !value.isPaused();
+            } else {
+                return false;
+            }
+        });
+    }
+
+    public LiveData<Integer> getPosition(String clipID) {
+        return getPositionForClip(clipID);
+    }
+
+    public void setPosition(String clipID, Integer newPosition) {
+        if (isCurrentPlayingClip(clipID, currentlyPlaying.getValue())) {
+            getMediaPlayer().seekTo(newPosition);
+        }
+
+        getPositionForClip(clipID).setValue(newPosition);
+    }
+
+    @Override
+    public void onCompletion(MediaPlayer mediaPlayer) {
+        clearClip();
+    }
+
+    public void background() {
+        clearClip();
+        releaseMediaPlayer();
+    }
+
+    @Override
+    protected void onCleared() {
+        clearClip();
+        releaseMediaPlayer();
+    }
+
+    private void clearClip() {
+        cancelPositionUpdates();
+        currentlyPlaying.setValue(null);
+    }
+
+    @NonNull
+    private MutableLiveData<Integer> getPositionForClip(String clipID) {
+        MutableLiveData<Integer> liveData;
+
+        if (positions.containsKey(clipID)) {
+            liveData = positions.get(clipID);
+        } else {
+            liveData = new MutableLiveData<>();
+            liveData.setValue(0);
+            positions.put(clipID, liveData);
+        }
+
+        return liveData;
+    }
+
+    private void schedulePositionUpdates() {
+        if (!scheduledDurationUpdates) {
+            scheduler.schedule(() -> {
+                CurrentlyPlaying currentlyPlaying = this.currentlyPlaying.getValue();
+
+                if (currentlyPlaying != null) {
+                    MutableLiveData<Integer> position = getPositionForClip(currentlyPlaying.clipID);
+                    position.postValue(getMediaPlayer().getCurrentPosition());
+                }
+            }, 500);
+            scheduledDurationUpdates = true;
+        }
+    }
+
+    private void cancelPositionUpdates() {
+        scheduler.cancel();
+        scheduledDurationUpdates = false;
+    }
+
+    private void releaseMediaPlayer() {
+        getMediaPlayer().release();
+        mediaPlayer = null;
+    }
+
+    private MediaPlayer getMediaPlayer() {
+        if (mediaPlayer == null) {
+            mediaPlayer = mediaPlayerFactory.create();
+            mediaPlayer.setOnCompletionListener(this);
+        }
+
+        return mediaPlayer;
+    }
+
+    private boolean isCurrentPlayingClip(String clipID, CurrentlyPlaying currentlyPlayingValue) {
+        return currentlyPlayingValue != null && currentlyPlayingValue.clipID.equals(clipID);
+    }
+
+    private void loadClip(String uri) {
+        try {
+            getMediaPlayer().reset();
+            getMediaPlayer().setDataSource(uri);
+            getMediaPlayer().prepare();
+        } catch (IOException ignored) {
+            throw new RuntimeException();
+        }
+    }
+
+    private enum ClipState {
```

Comment:
Seems unused.

Reply:
Good catch it is. Thought PMD/CheckStyle would have caught that! Will delete.